test/views/cancellation/CancelDeclarationViewSpec.scala

```diff
@@ -105,126 +101,23 @@ class CancelDeclarationViewSpec extends UnitViewSpec with CommonMessages with St
-          CancelDeclaration.form
-            .fillAndValidate(CancelDeclaration(Lrn("12345566++"), "123456789", "Some Description", NoLongerRequired.toString))
-        )
-
-        view must haveGovukGlobalErrorSummary
-        view must containErrorElementWithTagAndHref("a", "#functionalReferenceId")
-
-        view must containErrorElementWithMessageKey("cancellation.functionalReferenceId.error.specialCharacter")
-      }
-    }
-
-    "display mrn error" when {
-      "field is empty" in {
-        testView("", "Some Description", "mrn", "empty")
-      }
-
-      "field is entered but is too long" in {
-        val view = createView(
-          CancelDeclaration.form
-            .fillAndValidate(
-              CancelDeclaration(Lrn("1SA123456789012-1FSA1234567"), createRandomAlphanumericString(19), "Some Description", NoLongerRequired.toString)
-            )
-        )
-
-        view must haveGovukGlobalErrorSummary
-        view must containErrorElementWithTagAndHref("a", "#mrn")
-
-        view must containErrorElementWithMessageKey("cancellation.mrn.error.length")
-      }
-
-      "field is entered but is in the wrong format" in {
-        val view = createView(
-          CancelDeclaration.form
-            .fillAndValidate(CancelDeclaration(Lrn("1SA123456789012-1FSA1234567"), "1234567890123-", "Some Description", NoLongerRequired.toString))
-        )
-
-        view must haveGovukGlobalErrorSummary
-        view must containErrorElementWithTagAndHref("a", "#mrn")
-
-        view must containErrorElementWithMessageKey("cancellation.mrn.error.wrongFormat")
-      }
-
-      "field value is not found for the current user" in {
-        val view = createView(
-          CancelDeclaration.form
-            .fillAndValidate(
-              CancelDeclaration(Lrn("1SA123456789012-1FSA1234567"), "20GB46J8TMJ4RFGVA0", "Some Description", NoLongerRequired.toString)
-            )
-            .copy(errors = Seq(FormError.apply(mrnKey, "cancellation.mrn.error.denied")))
-        )
-
-        view must haveGovukGlobalErrorSummary
-        view must containErrorElementWithTagAndHref("a", "#mrn")
-
-        view must containErrorElementWithMessageKey("cancellation.mrn.error.denied")
-      }
-
-      "field value has already been submitted in a previous cancellation request" in {
-        val view = createView(
-          CancelDeclaration.form
-            .fillAndValidate(
-              CancelDeclaration(Lrn("1SA123456789012-1FSA1234567"), "20GB46J8TMJ4RFGVA0", "Some Description", NoLongerRequired.toString)
-            )
-            .copy(errors = Seq(FormError.apply(mrnKey, "cancellation.duplicateRequest.error")))
-        )
-
-        view must haveGovukGlobalErrorSummary
-        view must containErrorElementWithTagAndHref("a", "#mrn")
-
-        view must containErrorElementWithMessageKey("cancellation.duplicateRequest.error")
-      }
-    }
+    val view = createView(CancelDeclarationDescription.form.bind(Map[String, String]()))
```

Comment:
Missing test for how to display the `CancellationAlreadyRequested` error on the page